### PR TITLE
Add Config API (configs.csi.ceph.io)

### PR DESCRIPTION
**Pull request by nb-ohad**
# Describe what this PR does #

Adds new API named Config (configs.csi.ceph.io), defining the information Ceph CSI needs to consume storage from a Ceph cluster
The PR does not include CRD field validation rules.
This will be resolved in a future PR
Provide some context for the reviewer

## Is there anything that requires special attention ##
No

### CODE REVIEW

File: api/v1alpha1/config_types.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	RadosNamespace string `json:"radosNamespace,omitempty"`
+}
+
+// NfsConfigSpec cdefines the desired NFS configuration
+type NfsConfigSpec struct {
+}
+
+// ConfigSpec defines the desired state of Config
+type ConfigSpec struct {
+	CephClusterRef corev1.LocalObjectReference `json:"cephClusterRef,omitempty"`
```
Comment (Madhu-1) [inline api/v1alpha1/config_types.go:40]: @nb-ohad can you remove omitempty from this one, we need to have it set

Comment (nb-ohad) [reply api/v1alpha1/config_types.go:40]: Omit empty is a serialization directive it does not indicate that the field is optional or not, it just mean that the filed will not be serialized on write if the value is empty.
 
For an optional field, we need an `+optional` comment above the field, so OpenAPI will treat it as optional (which we don't have right now for anything).

But regardless I can remove the omitempty if you feel it is necessary

Comment (Madhu-1) [reply api/v1alpha1/config_types.go:40]: okay, it looks like i was not clear, i wanted to get it removed as its deadcode as we always set this value and its never going to be empty

Comment (nb-ohad) [reply api/v1alpha1/config_types.go:40]: Yep, I can do that. 
I usually set omitempty for all fields of my APIs to lower the footprint when manually marshaling (for example in unit test code)

Comment (nb-ohad) [reply api/v1alpha1/config_types.go]: done

**Review by Madhu-1 — APPROVED** [review]

**Pull request merged**